src/codegate/muxing/semantic_router.py

```diff
@@ -0,0 +1,129 @@
+import unicodedata
+import uuid
+
+import numpy as np
+import regex as re
+import structlog
+
+from codegate.config import Config
+from codegate.db import models as db_models
+from codegate.db.connection import DbReader, DbRecorder
+from codegate.inference.inference_engine import LlamaCppInferenceEngine
+
+logger = structlog.get_logger("codegate")
+
+
+class PersonaDoesNotExistError(Exception):
+    pass
+
+
+class SemanticRouter:
+
+    def __init__(self):
+        self._inference_engine = LlamaCppInferenceEngine()
+        conf = Config.get_config()
+        self._embeddings_model = f"{conf.model_base_path}/{conf.embedding_model}"
+        self._n_gpu = conf.chat_model_n_gpu_layers
+        self._persona_threshold = conf.persona_threshold
+        self._db_recorder = DbRecorder()
+        self._db_reader = DbReader()
```

Comment:
Not for this Pull Request: I'm starting to think DbRecorder should inherit from DBReader so we wouldn't need to call an explicit object for each particular operation. Having two explicit objects makes sense in a distributed system where you might have read replicas, but it's probably overkill for this single container system.